packages/cysync-core/src/dialogs/Inheritance/GoldPlanPurchase/Dialogs/Message/NomineePrivateMessage.tsx

```diff
@@ -33,7 +34,40 @@ export const NomineePrivateMessageInput = () => {
     isOnSummaryPage,
   } = useInheritanceGoldPlanPurchaseDialog();
 
-  const isNextDisabled = !personalMessage.trim() || !cardLocation.trim();
+  const [cardLocationError, setCardLocationError] = useState<string>('');
+  const [personalMessageError, setPersonalMessageError] = useState<string>('');
+
+  const isNextDisabled =
+    !personalMessage.trim() ||
+    !cardLocation.trim() ||
+    !cardLocationError ||
+    !personalMessageError;
+
+  useEffect(() => {
+    const validateFields = () => {
+      const cardLocationValidation = validateInputLanguage(cardLocation, lang);
+      const personalMessageValidation = validateInputLanguage(
+        personalMessage,
+        lang,
+      );
+
+      if (!cardLocationValidation.success) {
+        setCardLocationError(cardLocationValidation.error.errors[0].message);
+      } else {
+        setCardLocationError('');
+      }
+
+      if (!personalMessageValidation.success) {
+        setPersonalMessageError(
+          personalMessageValidation.error.errors[0].message,
+        );
+      } else {
+        setPersonalMessageError('');
+      }
+    };
+
+    validateFields();
+  }, [cardLocation, personalMessage]);
```

Comment:
Don't use regex calls on every stroke, use a debounce function for this.

Reply:
resolved